LoopKit/CGMManager.swift

```diff
@@ -77,4 +76,79 @@ public extension CGMManager {
     var appURL: URL? {
         return nil
     }
+
+    /// Convenience wrapper for notifying the delegate of deletion on the delegate queue
+    ///
+    /// - Parameters:
+    ///   - completion: A closure called from the delegate queue after the delegate is called
+    func notifyDelegateOfDeletion(completion: @escaping () -> Void) {
+        delegateQueue.async {
+            self.cgmManagerDelegate?.cgmManagerWantsDeletion(self)
+            completion()
+        }
+    }
+}
+
+
+
+public class CGMManagerDelegateWrapper {
+    public typealias Delegate = CGMManagerDelegate
```

Comment:
Looks like this class isn't actually specific to CGMManagerDelegate, and the pump manager equivalent is the same. You can make this generic, with a more descriptive name; something like `WeakSynchronized<Value>`. It'll require doing the AnyObject dance to attribute the underlying property as `weak`, but I think developing the reusable, generic component is worthwhile here.